From fc06241652dd76893d345811ade61afb4236cc13 Mon Sep 17 00:00:00 2001
From: Michel Pollet <buserror@gmail.com>
Date: Wed, 28 Nov 2012 16:03:29 +0000
Subject: [PATCH 04/11] imx23: Add olinuxino timings

Adds support for a 'dumb' 16bpp LCD panel for the
imx233 LQFP olinuxino board(s)

Signed-off-by: Claude Schwarz <claude.schwarz@gmail.com>
Signed-off-by: Michel Pollet <buserror@gmail.com>
---
 arch/arm/mach-mxs/mach-mxs.c |   31 +++++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/arch/arm/mach-mxs/mach-mxs.c b/arch/arm/mach-mxs/mach-mxs.c
index 4748ec5..33b3662 100644
--- a/arch/arm/mach-mxs/mach-mxs.c
+++ b/arch/arm/mach-mxs/mach-mxs.c
@@ -46,6 +46,26 @@ static struct fb_videomode mx23evk_video_modes[] = {
 	},
 };
 
+static struct fb_videomode oli_video_modes[] = {
+        {
+                .name           = "TX09D70",
+                .refresh        = 60,
+                .xres           = 240,
+                .yres           = 320,
+                .pixclock       = KHZ2PICOS(4965), /* picosecond (5.33 MHz) */
+                .left_margin    = 1,
+                .right_margin   = 32,
+                .upper_margin   = 1,
+                .lower_margin   = 0,
+                .hsync_len      = 5,
+                .vsync_len      = 1,
+                .sync           = FB_SYNC_DATA_ENABLE_HIGH_ACT |
+                                  FB_SYNC_DOTCLK_FAILING_ACT,
+
+        },
+};
+
+
 static struct fb_videomode mx28evk_video_modes[] = {
 	{
 		.name		= "Seiko-43WVF1G",
@@ -223,6 +243,15 @@ static void __init imx23_evk_init(void)
 	mxsfb_pdata.ld_intf_width = STMLCDIF_24BIT;
 }
 
+static void __init imx23_olinuxino_init(void)
+{
+        mxsfb_pdata.mode_list = oli_video_modes;
+        mxsfb_pdata.mode_count = ARRAY_SIZE(oli_video_modes);
+        mxsfb_pdata.default_bpp = 16;
+        mxsfb_pdata.ld_intf_width = STMLCDIF_16BIT;
+}
+
+
 static inline void enable_clk_enet_out(void)
 {
 	struct clk *clk = clk_get_sys("enet_out", NULL);
@@ -361,6 +390,8 @@ static void __init mxs_machine_init(void)
 		imx28_evk_init();
 	else if (of_machine_is_compatible("fsl,imx23-evk"))
 		imx23_evk_init();
+	else if (of_machine_is_compatible("olimex,imx23-olinuxino"))
+		 imx23_olinuxino_init();
 	else if (of_machine_is_compatible("denx,m28evk"))
 		m28evk_init();
 	else if (of_machine_is_compatible("bluegiga,apx4devkit"))
-- 
1.7.10.4

